src/libraries/System.Runtime.Extensions/tests/System/IO/RedundantSegmentsTests_Windows.cs

```diff
@@ -0,0 +1,832 @@
+            from p in DeviceAndExtendedPrefixes
+            select new object[] { p + Prefix_Windows_Drive_Root + t.Item1, p + Prefix_Windows_Drive_Root + t.Item2 };
+        public static IEnumerable<object[]> MemberData_Unqualified_Redundant_DriveRootless_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            select new object[] { Prefix_Windows_Drive_Rootless + t.Item1, Prefix_Windows_Drive_Rootless + t.Item3 };
+        public static IEnumerable<object[]> MemberData_Unqualified_Redundant_Prefix_DriveRootless_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            from p in DeviceAndExtendedPrefixes
+            select new object[] { p + Prefix_Windows_Drive_Rootless + t.Item1, p + Prefix_Windows_Drive_Rootless + t.Item4 };
+        public static IEnumerable<object[]> MemberData_Unqualified_Redundant_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            select new object[] { t.Item1, t.Item3 };
+        public static IEnumerable<object[]> MemberData_Unqualified_Redundant_DrivelessRoot_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            select new object[] { Prefix_Windows_Driveless_Root + t.Item1, Prefix_Windows_Driveless_Root + t.Item2 };
+        public static IEnumerable<object[]> MemberData_ServerShare_Redundant_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            select new object[] { ServerShare + t.Item1, ServerShare + t.Item2 };
+        public static IEnumerable<object[]> MemberData_UNC_Redundant_SingleDot =>
+            from t in TestPaths_Redundant_SingleDot
+            from p in DeviceAndExtendedPrefixes
+            select new object[] { p + UNCServerShare + t.Item1, p + UNCServerShare + t.Item2 };
+
+        #endregion
+
+        #region Double dot
+
+        private static readonly List<Tuple<string, string, string, string>> TestPaths_Redundant_DoubleDot = new List<Tuple<string, string, string, string>>
+        {
+            // The original and qualified strings must get the root string prefixed
+            // Original | Qualified | Unqualified | Device prefix
+            { @"..",     @"",    @"..",     @".." },
+            { @"..\",    @"",    @"..\",    @"..\" },
+            { @"..\..",  @"",    @"..\..",  @"..\.." },
+            { @"..\..\", @"",    @"..\..\", @"..\..\" },
+
+            { @"..\folder",          @"folder",      @"..\folder",  @"..\folder" },
+            { @"..\folder\",         @"folder\",     @"..\folder\", @"..\folder\" },
+            { @"..\folder\..",       @"",            @"..",         @"..\" },
+            { @"..\folder\..\",      @"",            @"..\",        @"..\" },
+            { @"..\folder\..\..",    @"",            @"..\..",      @"..\.." },
+            { @"..\folder\..\..\",   @"",            @"..\..\",     @"..\..\" },
+
+            { @"..\..\folder",           @"folder",      @"..\..\folder",   @"..\..\folder" },
```

Comment:
The device path should eat down to `..\folder` here as only the first `..` is the "volume".  Device paths are, by definition, always fully qualified.

Reply:
On that same note `\\?\C:A\bar.txt` the root is `\\?\C:A\`. Wanted to point that out so we don't get that confused with drive relative, which only happens in the `C:A` form.

Reply:
> _The device path should eat down to `..\folder` here as only the first `..` is the "volume". Device paths are, by definition, always fully qualified._

Thanks. If I'm understanding this correctly, it seems that my answer in the comment above will make sure this case gets covered.

> On that same note `\\?\C:A\bar.txt` the root is `\\?\C:A\`. Wanted to point that out so we don't get that confused with drive relative, which only happens in the `C:A` form.

Thanks for confirming. Yes, I found out about edge cases like `C:A` when I realized `Path.GetPathRoot()` was giving me some strange root lengths after the device prefix, but I read its code and understood the reason.

Reply:
To be fully explicit:

- `\\?\..\..\folder` -> `\\?\..\..\folder` (untouched)
- `\\.\..\..\folder` -> `\\.\..\folder` (`\\.\..\` is the root)
- `\\.\C:\..\folder` -> `\\.\C:\folder\` (for a more common root example)

Reply:
For the first example, I just made the adjustment in my last commits to ensure they are not modified. 

I see what you mean with your 2nd  example. I think my code may be able to handle that case right now, but I need to double check if I have unit tests for it.

The third case is handled for sure.

Reply:
Update: 2nd case is already handled by all unit tests with member data that contain the substring `Prefix_DriveRootless`.